contracts/vaults/Deposit.sol

```diff
@@ -50,9 +59,19 @@ contract Deposit is Base {
      * @param _pid The pool id
      */
     function claimReward(uint256 _pid) external {
-        UserInfo memory user = userInfo[_pid][msg.sender];
+        if (!poolInfos[_pid].committeeCheckedIn)
+            revert CommitteeNotCheckedInYet();
+        updatePool(_pid);
 
-        _claimReward(_pid, user);
+        UserInfo storage user = userInfo[_pid][msg.sender];
+        uint256 rewardPerShare = poolInfos[_pid].rewardPerShare;
+        if (user.shares > 0) {
+            uint256 pending = user.shares * rewardPerShare / 1e12 - user.rewardDebt;
+            user.rewardDebt = user.shares * rewardPerShare / 1e12;
```

Comment:
can move this line after `if (pending > 0)` to save some infinitesimal amount of gas :)